game-core/src/main/java/games/strategy/triplea/delegate/battle/casualty/CasualtySelector.java

```diff
@@ -254,6 +260,38 @@ private static void killAmphibiousFirst(final List<Unit> killed, final Collectio
     }
   }
 
+  /** Kill off the air units with the lowest movement */
+  private static void killLowestMovementAirUnit(
+      final Collection<Unit> killed, final Collection<Unit> targets) {
+    final Collection<Unit> airUnitsKilled =
+        killed.stream().filter(Matches.unitIsAir()).collect(Collectors.toList());
+    if (airUnitsKilled.isEmpty()) {
+      // no air units were killed so no need to kill lowest movement units
+      return;
+    }
+
+    final Map<UnitOwner, List<Unit>> targetsGroupedByOwnerAndType =
+        targets.stream().collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList()));
+    final Map<UnitOwner, List<Unit>> killedGroupedByOwnerAndType =
+        airUnitsKilled.stream().collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList()));
+
+    for (final Map.Entry<UnitOwner, List<Unit>> killedEntry :
```

Comment:
Minor, this for loop could perhaps benefit with a short summary comment to help guide the reader. Ideally a short outline of the algorithm or a few words on what we are doing. For example, the 'pop' at the end is not immediately obvious why it is needed, anything to help a reader understand this faster and not reverse engineer the intent is a win.